docs/topics/i18n/translation.txt

```diff
@@ -279,14 +279,60 @@ In a case like this, consider something like the following::
 
         a format specification for argument 'name', as in 'msgstr[0]', doesn't exist in 'msgid'
 
-.. note:: Plural form and po files
-
-    Django does not support custom plural equations in po files. As all
-    translation catalogs are merged, only the plural form for the main Django po
-    file (in ``django/conf/locale/<lang_code>/LC_MESSAGES/django.po``) is
-    considered. Plural forms in all other po files are ignored. Therefore, you
-    should not use different plural equations in your project or application po
-    files.
+.. _plural-forms:
+
+Plural Forms
+~~~~~~~~~~~~
+
+Django does not support multiple plural forms in catalogs. As all translation
+catalogs are merged, only the plural form in the main Django po file (located
+by default in ``django/conf/locale/<lang_code>/LC_MESSAGES/django.po``) is
+considered.
+
+To prevent inconsistencies and undesired results in translations, Django will
+not merge any catalog that contains a different plural form than the main one
+and will issue a warning about the conflicting catalog.
+
+This conflict may arise mostly in two situations:
+
+* when the main plural form for a language is updated in Django and your po
+  files were generated with a previous one, or
+
+* when including third-party translations with a different plural form.
+
+Django follows the standards provided by `Unicode
+<http://www.unicode.org/cldr/charts/latest/supplemental/language_plural_rules.html>`_
+for plural equations for each language and encourages to align with it.
```

Comment:
This is inaccurate. Django uses whatever Transifex did to convert CLDR plurals to gettext plural forms. Gettext uses integers for the calculation, while CLDR specifies plurals for floats as well. That leads to plural forms defined in CLDR, but unreachable in gettext. Transifex did not choose to skip those, so there can be plurals which will never be used. There are other conversions which handle this case correctly (I'm aware of https://github.com/php-gettext/Languages, but there might be others as well).

I've noticed this for Czech, but the issue might be in other languages as well. 

The plural form used by Transifex: `nplurals=4; plural=(n == 1 && n % 1 == 0) ? 0 : (n >= 2 && n "
"<= 4 && n % 1 == 0) ? 1: (n % 1 != 0 ) ? 2 : 3;`

Plural form usually used by gettext: `nplurals=3;plural=(n == 1) ? 0 : ((n >= 2 && n <= 4) ? 1 : 2)`

You can see that `many` from [CLDR](http://www.unicode.org/cldr/charts/latest/supplemental/language_plural_rules.html#cs) is missing from second formula. The Transifex formula has it, but it's unreachable because `n % 1 != 0` is never true and this the plural form never evaluates to 2. This brings inconsistency with rest of the software using gettext, additional effort for translators and brings exactly zero benefit.

Reply:
Django indeed uses plural forms from Transifex (and according to Transifex's documentation posted on the dev list, they follow Unicode). Following a standard does not mean implement it well.

The statement is about setting a policy about plural forms (no one objected it in the mailing list).

If the plural form is not an implementation of the CLDR specification, it should not be included in the catalogs distributed by Django.

If Django should maintain the plural forms independently of Transifex, I think that's another discussion. The PR is about ensuring consistency in plural forms and providing a way of modify them locally via LOCALE_ROOT.